common/main/cpp/native_c4listener.cc

```diff
@@ -64,57 +64,34 @@ bool litecore::jni::initC4Listener(JNIEnv *env) {
 
 static bool httpAuthCallback(C4Listener *ignore, C4Slice authHeader, void *context) {
     JNIEnv *env = nullptr;
-    jint getEnvStat = gJVM->GetEnv(reinterpret_cast<void **>(&env), JNI_VERSION_1_6);
-
-    jboolean ok = JNI_FALSE;
+    jint envState = attachJVM(&env, "httpAuth");
+    if ((envState != JNI_OK) && (envState != JNI_EDETACHED))
+        return false;
 
-    if (getEnvStat == JNI_OK) {
-        jstring _header = toJString(env, authHeader);
-        ok = env->CallStaticBooleanMethod(cls_C4Listener, m_C4Listener_httpAuthCallback, (jlong) context, _header);
-        if (_header != nullptr) env->DeleteLocalRef(_header);
-    } else if (getEnvStat == JNI_EDETACHED) {
-        if (attachCurrentThread(&env) == 0) {
-            ok = env->CallStaticBooleanMethod(
-                    cls_C4Listener,
-                    m_C4Listener_httpAuthCallback,
-                    (jlong) context,
-                    toJString(env, authHeader));
-            if (gJVM->DetachCurrentThread() != 0)
-                C4Warn("httpAuthCallback(): Failed to detach the current thread from a Java VM");
-        } else {
-            C4Warn("httpAuthCallback(): Failed to attaches the current thread to a Java VM");
-        }
+    jstring _header = toJString(env, authHeader);
+    jboolean ok = env->CallStaticBooleanMethod(cls_C4Listener, m_C4Listener_httpAuthCallback, (jlong) context, _header);
+    if (envState == JNI_EDETACHED) {
+        detachJVM("httpAuthCallback");
```

Comment:
You might consider passing the envState into this method to save on the repeated ifs throughout these files, but that's also not a big deal.